crates/data_components/src/duckdb.rs

```diff
@@ -351,11 +352,20 @@ impl Read for DuckDBTableFactory {
     ) -> Result<Arc<dyn TableProvider + 'static>, Box<dyn std::error::Error + Send + Sync>> {
         let pool = Arc::clone(&self.pool);
         let dyn_pool: Arc<DynDuckDbConnectionPool> = pool;
-        let table_provider = SqlTable::new(&dyn_pool, table_reference, None)
+        let table_provider = SqlTable::new("duckdb", &dyn_pool, table_reference, None)
             .await
             .map_err(|e| Box::new(e) as Box<dyn std::error::Error + Send + Sync>)?;
 
-        Ok(Arc::new(table_provider))
+        let table_provider = Arc::new(table_provider);
+
+        #[cfg(feature = "federation-experimental")]
+        let table_provider = Arc::new(
+            table_provider
+                .create_federated_table_provider()
+                .map_err(|e| Box::new(e) as Box<dyn std::error::Error + Send + Sync>)?,
+        );
```

Comment:
This is where the federated table is added. I've architected it such that it is optional - if the TableProvider is federated, the optimizer will use it - otherwise it ignores it and the current behavior is used.